code/modules/antagonists/nukeop/equipment/nuclearbomb.dm

```diff
@@ -501,16 +501,35 @@ This is here to make the tiles around the station mininuke change when it's arme
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 30, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	var/fake = FALSE
+	var/turf/lastlocation
+	var/loneop_event_weight_increase_chance
 
 /obj/item/disk/nuclear/Initialize()
 	. = ..()
 	if(!fake)
 		GLOB.poi_list |= src
+		START_PROCESSING(SSobj, src)
 
 /obj/item/disk/nuclear/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/stationloving, !fake)
 
+/obj/item/disk/nuclear/process()
+	if(fake)
+		STOP_PROCESSING(SSobj, src)
+		stack_trace("A fake nuke disk tried to call process(). Who the fuck and how the fuck")
+		return
+	var/turf/newturf = get_turf(src)
+	if(istype(newturf) && lastlocation == newturf)
+		if(loneop_event_weight_increase_chance > 5 && prob(loneop_event_weight_increase_chance - 5))
+			var/datum/round_event_control/operative/loneop = locate(/datum/round_event_control/operative) in SSevents.control
+			if(istype(loneop))
+				loneop.weight += 1
+		loneop_event_weight_increase_chance += 0.01
```

Comment:
Rather than increasing this per process, keep track of the last time the disk moved